src/Stratis.SmartContracts.Core/Receipts/PersistentReceiptRepository.cs

```diff
@@ -38,13 +38,33 @@ public Receipt Retrieve(uint256 hash)
         {
             using (DBreeze.Transactions.Transaction t = this.engine.GetTransaction())
             {
-                byte[] result = t.Select<byte[], byte[]>(TableName, hash.ToBytes()).Value;
+                return this.GetReceipt(t, hash);
+            }
+        }
 
-                if (result == null)
-                    return null;
+        /// <inheritdoc />
+        public IList<Receipt> RetrieveMany(IEnumerable<uint256> hashes)
```

Comment:
Nit: IEnumerable doesn't imply ordering, though it is required in this circumstance.